[video.js] Add requestPictureInPicture()

mattleff opened the pull request:
https://docs.videojs.com/player#requestPictureInPicture

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://docs.videojs.com/player#requestPictureInPicture

DangerBotOSS commented:
<!--
  0 failure: 
  0 warning: 
  
  1 markdown notices
  DangerID: danger-id-Danger;
-->



Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## video.js ([<kbd>unpkg</kbd>](https://unpkg.com/browse/video.js@latest/))
was missing the following properties:
1. The declaration doesn't match the JavaScript module 'video.js'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.

The most common way to resolve this error is to use 'export =' syntax.
To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
1. hooks_
1. hooks
1. hook
1. hookOnce

<details>
<summary>as well as these 52 other properties...</summary>
<p>removeHook, VERSION, options, getPlayers, getPlayer, getAllPlayers, players, getComponent, registerComponent, getTech, registerTech, use, middleware, browser, TOUCH_ENABLED, extend, mergeOptions, bind, registerPlugin, deregisterPlugin, plugin, getPlugins, getPlugin, getPluginVersion, addLanguage, log, createLogger, createTimeRange, createTimeRanges, formatTime, setFormatTime, resetFormatTime, parseUrl, isCrossOrigin, EventTarget, on, one, off, trigger, xhr, TextTrack, AudioTrack, VideoTrack, computedStyle, dom, url, mediaSources, MediaSource, URL, HlsHandler, HlsSourceHandler, Hls</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 67dfd84bf95e61c2f9dcfedf7db5b03de9cbfc38
</p>

peterblazejewicz commented:
Not sure about return type here, should the api be like:
```ts
player.requestPictureInPicture().then(pipWindow => // $ExpectType PictureInPictureWindow);
```
it returns an object with a shape of `PictureInPictureWindow` or promise gets rejected
```ts
PictureInPictureWindow {width: 384, height: 160, onresize: null}
```

mattleff commented:
@peterblazejewicz Thanks for the feedback! I've changed the return type and added `exitPictureInPicture()` as well. I added a `PictureInPictureWindow` type directly on the `videojs` namespace, but if there's a better place for it, or if it's better to use an inline type definition I can adjust as needed.

peterblazejewicz commented on types/video.js/index.d.ts:
```diff
@@ -3913,6 +3913,21 @@ declare namespace videojs {
     }
 
+    /**
+     * The `PictureInPictureWindow` interface represents an object able to programmatically obtain the `width` and `height` of the floating video window.
```
```suggestion
     * The `PictureInPictureWindow` interface represents an object able to programmatically obtain the `width` and `height` of the floating video window.
     * @see {@link https://developer.mozilla.org/en-US/docs/Web/API/PictureInPictureWindow}
```

peterblazejewicz reviewed (commented):
Please review a comment, otherwise ready to merge

peterblazejewicz reviewed (approved):
LGTM!
@mattleff thx!

mattleff commented:
Ready to merge
